pkg/storageops/storageops.go

```diff
@@ -0,0 +1,74 @@
+package storageops
+
+const (
+	// SetIdentifierNone is an default identifier to group all disks managed by
+	// storage ops
+	SetIdentifierNone = "None"
+)
+
+// Custom storage operation error codes.
+const (
+	_ = iota + 5000
+	ErrVolDetached
+	ErrVolInval
+	ErrVolAttachedOnRemoteNode
+)
+
+// StorageError error returned for storage operations
+type StorageError struct {
+	// Code is one of storage operation driver error codes.
+	Code int
+	// Msg is human understandable error message.
+	Msg string
+	// Instance provides more information on the error.
+	Instance string
+}
+
+// Ops interface to perform basic storage operations.
+type Ops interface {
+	// Name returns name of the storage operations driver
+	Name() string
+	// Create volume based on input template volume and also apply given labels.
+	Create(template interface{}, labels map[string]string) (interface{}, error)
+	// GetDeviceID returns ID/Name of the given device/disk
+	GetDeviceID(template interface{}) string
+	// Attach volumeID.
+	// Return attach path.
+	Attach(volumeID string) (string, error)
+	// Detach volumeID.
+	Detach(volumeID string) error
+	// Delete volumeID.
+	Delete(volumeID string) error
+	// FreeDevices returns free block devices on the instance
+	FreeDevices(blockDeviceMappings []interface{}, rootDeviceName string) ([]string, error)
```

Comment:
nit: add what blockDeviceMappings interface is

Reply:
done